coreneuron/io/nrn_setup.cpp

```diff
@@ -1026,47 +1043,125 @@ size_t model_size(void) {
+            size_data[11] += nb_nt;
+
+        }
+
     }
 
-#ifdef DEBUG
-    printf("%d netcon pointers %ld  nbyte=%ld\n", nrnmpi_myid, nccnt, nccnt * sizeof(NetCon*));
-#endif
     nbyte += nccnt * sizeof(NetCon*);
     nbyte += output_presyn_size();
     nbyte += input_presyn_size();
 
+    nbyte += nrnran123_instance_count() * nrnran123_state_size();
+
 #ifdef DEBUG
+    printf("%d netcon pointers %ld  nbyte=%ld\n", nrnmpi_myid, nccnt, nccnt * sizeof(NetCon*));
     printf("nrnran123 size=%ld cnt=%ld nbyte=%ld\n", nrnran123_state_size(),
            nrnran123_instance_count(), nrnran123_instance_count() * nrnran123_state_size());
+    printf("%d total bytes %ld\n", nrnmpi_myid, nbyte);
+#endif
+    if (detailed_report) {
+        size_data[12] = nbyte;
+#if NRNMPI
+        MPI_Allreduce(&size_data[0],
+                      &global_size_data_min[0],
+                      13,
+                      MPI_UNSIGNED_LONG_LONG,
+                      MPI_MIN,
+                      MPI_COMM_WORLD);
+        MPI_Allreduce(&size_data[0],
+                      &global_size_data_max[0],
+                      13,
+                      MPI_UNSIGNED_LONG_LONG,
+                      MPI_MAX,
+                      MPI_COMM_WORLD);
+        MPI_Allreduce(&size_data[0],
+                      &global_size_data_sum[0],
+                      13,
+                      MPI_UNSIGNED_LONG_LONG,
+                      MPI_SUM,
+                      MPI_COMM_WORLD);
+        for (int i = 0; i < 13; i++) {
+            global_size_data_avg[i] = global_size_data_sum[i] / float(nrnmpi_numprocs);
+        }
+#else
+        global_size_data_max = size_data;
+        global_size_data_min = size_data;
+        for (int i = 0; i < 13; i++) {
+            global_size_data_avg[i] = size_data[i];
+        }
 #endif
+        // now print the collected data:
+        if (nrnmpi_myid == 0) {
+          printf("Memory size information for all NrnThreads per rank\n");
+          printf("------------------------------------------------------------------\n");
+          printf("%22s %12s %12s %12s\n","field", "min", "max", "avg");
+          printf("%22s %12ld %12ld %15.2f\n","ncell",            global_size_data_min[0], global_size_data_max[0], global_size_data_avg[0]);
+          printf("%22s %12ld %12ld %15.2f\n","end",              global_size_data_min[1], global_size_data_max[1], global_size_data_avg[1]);
+          printf("%22s %12ld %12ld %15.2f\n","nmech",            global_size_data_min[2], global_size_data_max[2], global_size_data_avg[2]);
+          printf("%22s %12ld %12ld %15.2f\n","_ndata",           global_size_data_min[3], global_size_data_max[3], global_size_data_avg[3]);
+          printf("%22s %12ld %12ld %15.2f\n","_nidata",          global_size_data_min[4], global_size_data_max[4], global_size_data_avg[4]);
+          printf("%22s %12ld %12ld %15.2f\n","_nvdata",          global_size_data_min[5], global_size_data_max[5], global_size_data_avg[5]);
```

Comment:
I kind of like them to be separate since _nvdata indicates use of arbitrary size Vector data (and maybe some space for Random123). If I remember correctly, the _nidata are mostly integer indices into _ndata (and sometimes just an integer with special semantics).